### PR TITLE
Create component for mentor application view

### LINKED ISSUE #27: Implement the admin dashboard - mentor application view
### Description
As part of this issue, we need to create the admin dashboard - mentor application view, which should closely match the provided mockup. This will provide the admins with a way to view an application of a mentor and approve/ reject the mentor.

### Tasks
1. Set up the basic React components as required.
1. Implement the layout and design of the admin dashboard - mentor application view, following the provided mockup.
1. Integrate the necessary components and assets to achieve the desired functionality and visual appearance.
1. Ensure responsiveness and compatibility across different devices and screen sizes.

Mockup: https://www.figma.com/file/nehm2ff8aR7NkPerg39fK9/ScholarX-V2-%7C-Wireframes?type=design&node-id=1974-3807&mode=design&t=zBGgiYyW1foSirHq-4

image

Note: You may want to mock the API refer to the [API doc - Get a specific mentor's details](https://documenter.getpostman.com/view/27421496/2s93m1a4ac#ec8d5137-ea84-43db-8c60-e0c2c3039d2a) and [API doc - Approve/ Reject a mentor](https://documenter.getpostman.com/view/27421496/2s93m1a4ac#fbe79b03-902d-40dc-b391-94b937df7ee5) to see the relevant details.

### Acceptance Criteria
- The user interface closely resembles the provided mockup.
- All required sections, elements, and content are present and displayed correctly.
- The interface is responsive and works well on various devices and screen sizes.
- The implementation follows best practices and coding standards.
- Unit tests are written.

### PR DESCRIPTION
**Pull request by Piumal1999**
## Purpose

The purpose of this PR is to fix #27 

## Goals

Implementing the mentor application (profile) view

## Approach

- Created the storybook component and implemented it in a page

## Screenshots

Screenshot 2024-05-11 at 04 13 31

https://github.com/sef-global/scholarx-frontend/assets/27498587/a23b235e-6186-43d5-816a-0b669810ee58

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

## Related PRs
 
N/A

## Test environment
 
N/A

## Learning

N/A

### CODE REVIEW

**Comment by Piumal1999** [comment]
Note: This PR contains only the design with mock data.

Button actions, mentor state options, data validations are not done yet.

File: src/components/Dashboard/scenes/ManageMentorApplications/ManageMentorApplications.tsx [hunk 1]
```diff
@@ -139,7 +139,13 @@ export const ManageMentorApplications: React.FC = () => {
           <tbody className="bg-white">
             {filteredMentorsByName.map((mentor) => (
               <tr key={mentor.uuid}>
-                <td className="px-6 py-2 whitespace-no-wrap border-b border-gray-200 w-1/4">
+                <td
+                  className="px-6 py-2 whitespace-no-wrap border-b border-gray-200 w-1/4"
+                  onClick={() => {
+                    window.open(`/dashboard/mentor-application/${mentor.uuid}`);
+                  }}
+                  style={{ cursor: 'pointer' }}
```
Comment (anjula-sack) [inline src/components/Dashboard/scenes/ManageMentorApplications/ManageMentorApplications.tsx]: You can use the Link component from the router dom to navigate. Also `cursor-pointer` tailwind class to make it a cursor. But wrapping it with the link would do it by default/

File: src/components/MentorApplication/MentorApplication.component.tsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+    if (mentor != null) {
+      onApprove(mentor.uuid);
+    }
+  };
+
+  const handleReject = () => {
+    if (mentor != null) {
+      onReject(mentor.uuid);
+    }
+  };
```
Comment (anjula-sack) [inline src/components/MentorApplication/MentorApplication.component.tsx]: Can you integrate the API here? it's `admin/mentors/:mentorId/status` with the status on the body

File: src/components/MentorApplication/MentorApplication.component.tsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+              alt=""
+              className="w-28 rounded-full object-fill"
+            />
+            <div className="ml-5">
+              <div className="flex items-center space-x-3">
+                <span className="text-2xl font-semibold">
+                  {mentor.application.firstName} {mentor.application.lastName}
+                </span>
+                <span className="whitespace-nowrap rounded-full bg-blue-100 px-2.5 py-0.5 text-sm text-blue-700">
+                  Pending
```
Comment (anjula-sack) [inline src/components/MentorApplication/MentorApplication.component.tsx]:
```suggestion
                  {mentor.state}
```

**Review by anjula-sack — CHANGES_REQUESTED** [review]

File: src/components/Dashboard/scenes/ManageMentorApplications/ManageMentorApplications.tsx (hunk 1 above)
Comment (Piumal1999) [reply src/components/Dashboard/scenes/ManageMentorApplications/ManageMentorApplications.tsx]: Updated with a24c510

File: src/components/MentorApplication/MentorApplication.component.tsx (hunk 3 above)
Comment (Piumal1999) [reply src/components/MentorApplication/MentorApplication.component.tsx]: Updated with a24c510

**Review by anjula-sack — COMMENTED** [review]
@Piumal1999 There are some merge conflicts. can you take a look at them as well?

**Comment by Piumal1999** [comment]
Added the requested changes. Couldn't test with the backend yet

**Comment by Piumal1999** [comment]
Added the requested changes. Couldn't test with the backend yet

File: src/hooks/useMentor.ts [hunk 4]
```diff
@@ -16,10 +18,28 @@ const useMentor = (mentorId: string | undefined) => {
     },
   });
 
+  const { mutate: changeState } = useMutation({
+    mutationFn: async (newState: string) => {
+      if (mentorId != null) {
+        const { data } = await axios.put(
+          `${API_URL}/mentors/${mentorId}/state`,
```
Comment (anjula-sack) [inline src/hooks/useMentor.ts]:
```suggestion
          `${API_URL}/admin/mentors/${mentorId}/state`,
```

File: src/pages/Dashboard/scenes/ManageMentorApplications/MentorApplicationPage.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+              className="h-4 w-4"
+              viewBox="0 0 20 20"
+              fill="currentColor"
+            >
+              <path
+                fillRule="evenodd"
+                d="M7.293 14.707a1 1 0 010-1.414L10.586 10 7.293 6.707a1 1 0 011.414-1.414l4 4a1 1 0 010 1.414l-4 4a1 1 0 01-1.414 0z"
+                clipRule="evenodd"
+              />
+            </svg>
```
Comment (anjula-sack) [inline src/pages/Dashboard/scenes/ManageMentorApplications/MentorApplicationPage.tsx]: Let's move svg's to separate folder

**Review by anjula-sack — CHANGES_REQUESTED** [review]

**Comment by anjula-sack** [comment]
Any updates @Piumal1999?

**Review by anjula-sack — APPROVED** [review]
Thank you! @Piumal1999

**Pull request merged**